diff --git a/test/e2e/custom_tool_test.go b/test/e2e/custom_tool_test.go
index 2afc469fc..75e8500a3 100644
--- a/test/e2e/custom_tool_test.go
+++ b/test/e2e/custom_tool_test.go
@@ -193,8 +193,7 @@ func startCMPServer(configFile string) {
        os.Setenv("ARGOCD_PLUGINSOCKFILEPATH", pluginSockFilePath)
        if _, err := os.Stat(pluginSockFilePath); os.IsNotExist(err) {
                // path/to/whatever does not exist
-               err := os.Mkdir(pluginSockFilePath, 0700)
-               CheckError(err)
+               FailOnErr(Run("", "mkdir", "-p", pluginSockFilePath))
        }
        FailOnErr(RunWithStdin("", "", "../../dist/argocd", "--config-dir-path", configFile))
 }
